sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessorPartitionOwnership.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Messaging.EventHubs.Primitives
+{
+    /// <summary>
+    ///   Contains all the information needed to describe the status of the owner of a partition.  It's used by
+    ///   an <c>EventProcessor</c> for cooperative distribution of processing for the associated Event Hub.
+    /// </summary>
+    ///
+    /// <seealso href="https://www.nuget.org/packages/Azure.Messaging.EventHubs.Processor" />
+    ///
+    public class EventProcessorPartitionOwnership
+    {
+        /// <summary>
+        ///   The fully qualified Event Hubs namespace this ownership is associated with.  This
+        ///   is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </summary>
+        ///
+        public string FullyQualifiedNamespace { get; set; }
+
+        /// <summary>
+        ///   The name of the specific Event Hub this ownership is associated with, relative
+        ///   to the Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public string EventHubName { get; set; }
+
+        /// <summary>
+        ///   The name of the consumer group this ownership is associated with.
+        /// </summary>
+        ///
+        public string ConsumerGroup { get; set; }
+
+        /// <summary>
+        ///   The identifier of the associated <c>EventProcessor</c> instance.
+        /// </summary>
+        ///
+        public string OwnerIdentifier { get; set; }
+
+        /// <summary>
+        ///   The identifier of the Event Hub partition this ownership is associated with.
+        /// </summary>
+        ///
+        public string PartitionId { get; set; }
+
+        /// <summary>
+        ///   The date and time, in UTC, that the last update was made to this ownership.
+        /// </summary>
+        ///
+        public DateTimeOffset LastModifiedTime { get; set; }
+
+        /// <summary>
+        ///   The version metadata needed to update this ownership.
+        /// </summary>
+        ///
+        public string Version { get; set; }
```

Comment:
Just curious on the rationale of the name change away from `ETag` when we are still semantically dealing with it as an `ETag`.

Reply:
The SDK architecture board debated it and ultimately decided that there's too much of an HTTP connotation with `ETag` and we should use something more general for basic optimistic concurrency guards.

Reply:
Interesting - Is that naming going to be applied throughout the storage and Cosmos client libraries as well?

Reply:
To my knowledge, no.  This feedback was specific to the Event Processor&lt;T&gt; design in .NET, where there is no true concept of durable storage, just a set of abstract methods that are required to return data.    Even the other Event Hubs libraries are using ETag in public types, as they have pluggable storage via an abstraction like our internal `StorageManager` type.

Some of the points of discussion revolved around in-memory storage and local file stores where there's not really any external communication required.     If I'm being honest, there were strong opinions on both sides of the discussion and I wasn't able to fully follow some of the overlapping conversations as they took place., so I may be giving you a fragmented picture of the overall context.

Reply:
Thanks for the detailed context, that helps!
I'll just add my 2 cents that `Version`, for me, implies some ordering while `ETag` has become a commonly accepted representation of an optimistic concurrency value with "ifMatch" semantics agnostic of transport.